Bull is now a Crusher strain (ft. a minor rework)

lKiseki opened the pull request:
# About The Pull Request
Bull is no longer a Tier 2 caste, and is instead a strain of the Crusher, a Tier 3 caste.
By design, it is meant to be a much faster version of its original form, more focused on attacking rather than sieging.

### PLEASE TEST MERGE THIS FIRST I DO NOT TRUST MY CODE
### ALSO I NEED TO TEST BALANCE
## Changes
- Melee damage increased from 24 to 25.
- Attack delay reduced from 8 to 7.
- Maximum plasma increased from 340 to 520.
- Plasma gain increased from 24 to 40.
- Sunder reduction increased from 10% to 15%.
- Armor values increased;
  - Before: 50 melee, 55 bullet, 55 laser, 55 energy, 20 bomb, 35 bio, 50 fire, 35 acid
  - After: 65 melee, 65 bullet, 65 laser, 65 energy, 65 bomb, 65 bio, 65 fire, 65 acid
- Gains stagger resistance.
### Bull Charge
- No longer has three types of charges. It has all been made into a single charge type that combines Headbutt and Gore. 
  - Plow did nothing so we don't talk about it.
- No longer injects Ozelomelyn on a successful hit.
- Now vulnerable to razorwire. Charging into it will entangle the Bull.
- No, it didn't get structural damage. Just for the record.
### (NEW) Bull Stomp
- Highly comparable to Crusher's stomp.
- Costs 50 plasma instead of 100.
- Has a cooldown of 5 seconds, instead of 20.
- Shockwave effect does not knock down. Instead, it staggers and slows.
### (NEW) Scorched Earth
_Costs 250 plasma. 3 minute cooldown._

https://github.com/user-attachments/assets/9c9d7bc3-811b-4773-9bf7-51d638b7ed57

- This is the Bull's new **primordial ability.**
- On use, grants the Bull three charges of the ability.
- Each charge can be used to dash towards a targeted location, leaving a trail of super charged plasma behind.
- This trail will deal heavy damage (except to you and your allies, of course), while inflicting visual blurriness, stagger, and slowdown, to anyone that is caught standing on it.
- All damage dealt by this ability has **full penetration**, bypassing most forms of armor and protection.
- A charge must be used every 6 seconds; otherwise, the ability will cancel itself.

##
#### Other minor changes
- Various code changes to remove Bull as a caste, and add it as a Crusher strain instead.
- Removes unnecessary mob vars and rolls them into their respective abilities.


# Why It's Good For The Game
Finally saves Bull from its weird status in limbo, by making it into a strain of the caste it most closely resembled, and thereby having an excuse to get new things and cool changes that should make it a more viable choice, compared to what it was before.

# Changelog
:cl: Lewdcifer
del: Removes Bull as a Tier 2 caste.
add: Adds Bull as a strain of Crusher, the Tier 3 caste.
add: Only has one charge type, that includes Headbutt and Gore's bonuses, all in one.
balance: Bull charge no longer inflicts Ozelomelyn.
balance: Bull charge is now vulnerable to razorwire. Charging into it will entangle them.
add: Bull now gets its own version of Crusher's stomp.
balance: Bull's stomp costs 50 plasma instead of 100, and has a cooldown of 5 seconds instead of 20.
balance: Shockwave effect does not knock down. Instead, it staggers and slows.
add: Replaces Bull's primordial ability with Scorched Earth.
add: On use, grants three charges of the ability. Each charge can be used to dash towards a targeted location, leaving a trail of super charged plasma behind, which deals heavy damage and various debuffs to anyone standing on it.
add: A charge must be used every 6 seconds; otherwise, the ability will cancel itself. Be quick!
refactor: Various code changes to remove Bull as a caste, and add it as a Crusher strain instead.
code: Removes unnecessary mob vars and bakes them into their respective abilities.
/:cl:

jwkDT commented:
wow this looks like an absurd upgrade to crusher minus the sieging abilities

Lauwma commented:
> * Attack speed increased from 8 to 7 (yes, this is faster).

Attack delay reduced from 8 to 7 (higher attack speed)

despong commented:
![image](https://github.com/user-attachments/assets/d283882f-f048-4066-a07f-bbd0d540e4d9)

Mannybrado commented:
if it doesnt inject ozy then its just headbutt charge

Technical-l commented:
> wow this looks like an absurd upgrade to crusher minus the sieging abilities

you lose armor, siege, and fishing abilities for combat ones. seems ok to me and we may finally see bull having a place in tgmc

Dreathtil commented:
Something Fucked
Slowdown and stagger lasts like 30 seconds, can chain together and stack
Just from the first TM, and all I can say is
Way too fast for how tanky it is. Not really being slowdownable+staggerable means it has minimal risk back and forth charging since it begins charging quicker.

lKiseki commented:
Changes since last test merge:
- Changes to Bull's charge reverted.
- Renamed the Bull's Stomp ability to "Bull's Stomp" (previously just "Stomp").
- Fixed debuff durations in Bull's Stomp and Scorched Earth.
- Refactored Scorched Earth a bit. Should be usable now.
  - Tile damage increased from 10 to 15.
  - Added a visual indicator for when the ability is active.
- Bull can now be entangled in razorwire if it charges into it.

lKiseki commented:
**Changes since last test merge:**
- Maximum health reduced from 445 to 400.
- Bull is now immune to stagger (for real this time).
- Bull's Stomp cooldown increased from 5s to 10s.
- Bull's Stomp damage reduced from 60 to 50.
- Bull's Stomp paralyze duration reduced from 3s to 2s.
- Bull's Stomp stagger and slowdown stacks reduced from 3 to 1.5.
- Scorched Earth's dashing range reduced from 7 to 6.
- Scorched Earth can no longer bypass defensive structures such as barricades.

Kuro020 reviewed (changes requested):
Dont merge until the primo vfx is changed
